common/static/common/js/discussion/discussion_router.js

```diff
@@ -129,9 +128,7 @@
             DiscussionRouter.prototype.navigateToThread = function(thread_id) {
                 var thread;
                 thread = this.discussion.get(thread_id);
-                return this.navigate("" + (thread.get("commentable_id")) + "/threads/" + thread_id, {
-                    trigger: true
-                });
+                return this.showThread(thread);
```

Comment:
Also, it seems like with this change, the URL is not going to get updated when the user clicks on the thread.